### PR TITLE
prevent phantom buildings from crashing bot deployment

### LINKED ISSUE #2609: Princess AI cant deploy or move units in custom map.
[Stand alone megamek dev version 47.16, Windows 64bit ,Java version 1.8.0_271, player name BiggestBoss]

On a custom lunar tileset 40x40 map i have experianced an issue with the princess ai not deploying units except turrets on building and wall tiles.
I then tried connecting as the second player myself and deployed by hand,then disconnected and replaced my second instance of the game with the ai in hopes that it was just a deploy issue but then it turned out that even with the units deployed the ai would not make any moves.I will provide the megameklog document as well as the map and save in hopes that the issue can be found.

[bug log.txt](https://github.com/MegaMek/megamek/files/5888809/bug.log.txt)
[bug.sav.gz](https://github.com/MegaMek/megamek/files/5888813/bug.sav.gz)
[Lunar port base.zip](https://github.com/MegaMek/megamek/files/5888832/Lunar.port.base.zip)

### PR DESCRIPTION
**Pull request by NickAragua**
Fixes #2609

This error was caused by the presence of a building terrain type in a hex, but no CF. Likely this was due to a malformed map definition.

### CODE REVIEW

**Review by Windchild292 — APPROVED** [review]

File: megamek/src/megamek/common/pathfinder/BoardClusterTracker.java [hunk 1]
```diff
@@ -365,7 +366,13 @@ private boolean buildingPlowThroughRequired(Entity entity, MovementType relevant
         if (!hex.containsTerrain(Terrains.BLDG_CF) && !hex.containsExit(Terrains.FUEL_TANK_CF)) {
             return false;
         } else if (relevantMovementType == MovementType.Walker) {
-            int buildingCF = board.getBuildingAt(coords).getCurrentCF(coords);
+            Building building = board.getBuildingAt(coords);
```
Comment (Windchild292) [inline megamek/src/megamek/common/pathfinder/BoardClusterTracker.java:369]: Could make this final

Comment (NickAragua) [reply megamek/src/megamek/common/pathfinder/BoardClusterTracker.java:369]: Does that help performance at all, or is it just a style preference?

Comment (Windchild292) [reply megamek/src/megamek/common/pathfinder/BoardClusterTracker.java:369]: The compiler might be able to improve the performance and it notes the variable won't change after being set.

Comment (sixlettervariables) [reply megamek/src/megamek/common/pathfinder/BoardClusterTracker.java]: The perf benefits only apply to values which can be statically known (e.g. effective constants), as in that case results can be pre-computed, memoized, or omitted. There exists the possibility the JVM could elide a barrier when reading from a _final class member_, but my understanding is that is not implemented. In all other cases it is a nice-to-have to signal intent to other programmers.

Consider for local variables that if the compiler can determine if you violate `final`, it also knows if it is _effectively_ `final`.

**Pull request merged**